### PR TITLE
[Assistant Builder] Animate instructions suggestions

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
New suggestions animate by appearing from the right. Also fixes an issue in which stale suggestions were not pushed into history (they remained visible, sometimes preventing better suggestion to be visible) 
cc @Duncid , will first rebase this PR on yours #4893 before merging

Screencast from 2024-04-29 11-48-09

Risk
---
None (blast radius limited to instructions suggestions + gated)

Deploy Plan
---
Deploy front

### CODE REVIEW

File: front/components/assistant_builder/InstructionScreen.tsx [hunk 1]
```diff
@@ -448,27 +449,27 @@ function Suggestions({
...
-                  <div key={index} className="w-[320px] transition-all">
-                    <ContentMessage size="sm" title="" variant="sky">
-                      {suggestion}
-                    </ContentMessage>
-                  </div>
+              <div className="flex w-max">
+                {suggestions.map((suggestion) => (
+                  <AnimatedSuggestion
+                    suggestion={suggestion}
+                    key={suggestion}
```
Comment (flvndvd) [inline front/components/assistant_builder/InstructionScreen.tsx:471]: Can we use the index here? so we keep to a small size?

File: front/components/assistant_builder/InstructionScreen.tsx [hunk 2]
```diff
@@ -532,6 +562,18 @@ function mergeSuggestions(
       0,
       VISIBLE_SUGGESTIONS_NUMBER
     );
+    // shift right deprecated suggestions
+    for (const suggestion of visibleSuggestions) {
+      if (!bestRankedSuggestions.includes(suggestion)) {
+        const shiftedElement = mergedSuggestions.splice(
+          mergedSuggestions.indexOf(suggestion),
+          1
+        )[0];
```
Comment (flvndvd) [inline front/components/assistant_builder/InstructionScreen.tsx:571]: `nit`
```suggestion
        const [shiftedElement] = mergedSuggestions.splice(
          mergedSuggestions.indexOf(suggestion),
          1
        );
```

File: front/components/assistant_builder/InstructionScreen.tsx [hunk 3]
```diff
@@ -532,6 +562,18 @@ function mergeSuggestions(
       0,
       VISIBLE_SUGGESTIONS_NUMBER
     );
+    // shift right deprecated suggestions
+    for (const suggestion of visibleSuggestions) {
```
Comment (flvndvd) [inline front/components/assistant_builder/InstructionScreen.tsx:566]: To make this ordering logic clearer, WDYT about:
```
// Filter out the deprecated suggestions and append them after the best suggestions
mergedSuggestions = [
  ...mergedSuggestions.filter(suggestion => bestRankedSuggestions.includes(suggestion)),
  ...mergedSuggestions.filter(suggestion => !bestRankedSuggestions.includes(suggestion))
];
```

**Review by flvndvd — APPROVED** [review]
LGTM

File: front/components/assistant_builder/InstructionScreen.tsx (hunk 1 above)
Comment (philipperolet) [reply front/components/assistant_builder/InstructionScreen.tsx]: went with a hash, so same key for same content

**Pull request merged**